pkg/solana/txm/metrics.go

```diff
@@ -0,0 +1,209 @@
+	}, []string{"chainID"})
+)
+
+type solTxmMetrics struct {
+	metrics.Labeler
+	chainID string
+
+	// successful transactions
+	successTxs   metric.Int64Counter
+	finalizedTxs metric.Int64Counter
+
+	// inflight transactions
+	pendingTxs metric.Int64Gauge
+
+	// error cases
+	errorTxs          metric.Int64Counter
+	revertTxs         metric.Int64Counter
+	rejectTxs         metric.Int64Counter
+	dropTxs           metric.Int64Counter
+	simRevertTxs      metric.Int64Counter
+	simOtherTxs       metric.Int64Counter
+	dependencyFailTxs metric.Int64Counter
+}
+
+func NewSolTxmMetrics(chainID string) (*solTxmMetrics, error) {
+	m := beholder.GetMeter()
+	var err error
+
+	successTxs, err := m.Int64Counter("solana_txm_tx_success")
+	if err != nil {
+		return nil, fmt.Errorf("failed to register solana success txs: %w", err)
+	}
+
+	finalizedTxs, err := m.Int64Counter("solana_txm_tx_finalized")
+	if err != nil {
+		return nil, fmt.Errorf("failed to register solana finalized txs: %w", err)
+	}
+
+	pendingTxs, err := m.Int64Gauge("solana_txm_tx_pending")
+	if err != nil {
+		return nil, fmt.Errorf("failed to register solana pending txs: %w", err)
+	}
+
+	errorTxs, err := m.Int64Counter("solana_txm_tx_error")
+	if err != nil {
+		return nil, fmt.Errorf("failed to register solana error txs: %w", err)
+	}
+
+	revertTxs, err := m.Int64Counter("solana_txm_tx_error_revert")
+	if err != nil {
+		return nil, fmt.Errorf("failed to register solana revert txs: %w", err)
+	}
+
+	rejectTxs, err := m.Int64Counter("solana_txm_tx_error_reject")
+	if err != nil {
+		return nil, fmt.Errorf("failed to register solana reject txs: %w", err)
+	}
+
+	dropTxs, err := m.Int64Counter("solana_txm_tx_error_drop")
+	if err != nil {
+		return nil, fmt.Errorf("failed to register solana drop txs: %w", err)
+	}
+
+	simRevertTxs, err := m.Int64Counter("solana_txm_tx_error_sim_revert")
+	if err != nil {
+		return nil, fmt.Errorf("failed to register solana sim revert txs: %w", err)
+	}
+
+	simOtherTxs, err := m.Int64Counter("solana_txm_tx_error_sim_other")
+	if err != nil {
+		return nil, fmt.Errorf("failed to register solana sim other txs: %w", err)
+	}
+
+	dependencyFailTxs, err := m.Int64Counter("solana_txm_tx_error_dependency")
+	if err != nil {
+		return nil, fmt.Errorf("failed to register solana dependency fail txs: %w", err)
+	}
+
+	return &solTxmMetrics{
+		chainID: chainID,
+		Labeler: metrics.NewLabeler().With("chainID", chainID),
+
+		successTxs:        successTxs,
+		finalizedTxs:      finalizedTxs,
+		pendingTxs:        pendingTxs,
+		errorTxs:          errorTxs,
+		revertTxs:         revertTxs,
+		rejectTxs:         rejectTxs,
+		dropTxs:           dropTxs,
+		simRevertTxs:      simRevertTxs,
+		simOtherTxs:       simOtherTxs,
+		dependencyFailTxs: dependencyFailTxs,
+	}, nil
+}
+
+func (m *solTxmMetrics) GetOtelAttributes() []attribute.KeyValue {
+	otelLabels := beholder.OtelAttributes(m.Labels).AsStringAttributes()
+	return otelLabels
```

Comment:
nit: could we just return it directly
```suggestion
	return beholder.OtelAttributes(m.Labels).AsStringAttributes()
```